setupTest.js

```diff
@@ -1 +1,22 @@
-jest.setTimeout(240000);
+/* eslint-disable */
+const rimraf = require('rimraf');
+const { join } = require('path');
+
+jest.setTimeout(300000);
+
+jasmine.getEnv().addReporter({
+    specStarted: (result) => (jasmine.currentTest = result),
+    specDone: (result) => (jasmine.currentTest = result),
+});
+
+beforeEach((done) => {
+    rimraf(join(jasmine.testPath, `../bin/*`), () => {
+        done();
```

Comment:
Jest will only wait for the first `done` call. Which is why I recommended [`del`](https://www.npmjs.com/package/del) instead, so you can do

```js
await del([
  join(jasmine.testPath, `../bin/*`),
  join(jasmine.testPath, `../binary/*`),
  join(jasmine.testPath, `../dist/*`),
]);
```

Also note that you don't want `join` as windows slashes are not valid glob file separators. You can use `path.unix.join`, tho (as noted in `del`'s readme)

Reply:
or use `util.promisify(rimraf)` and `Promise.all`